src/librbd/api/Group.cc

```diff
@@ -1281,6 +1281,35 @@ int Group<I>::snap_rollback(librados::IoCtx& group_ioctx,
     return -ENOENT;
   }
 
+  if (group_snap->state != cls::rbd::GROUP_SNAPSHOT_STATE_COMPLETE) {
+    lderr(cct) << "group snapshot is not complete" << dendl;
+    return -EINVAL;
+  }
+
+  std::vector<cls::rbd::GroupImageSpec> rollback_images;
+  for (const auto& snap : group_snap->snaps) {
+    rollback_images.emplace_back(snap.image_id, snap.pool);
+  }
+
+  std::vector<cls::rbd::GroupImageStatus> images;
+  r = group_image_list(group_ioctx, group_name, &images);
+  if (r < 0) {
+    return r;
+  }
+
+  std::vector<cls::rbd::GroupImageSpec> current_images;
+  for (const auto& image : images) {
+    if (image.state == cls::rbd::GROUP_IMAGE_LINK_STATE_ATTACHED) {
+      current_images.push_back(image.spec);
+    }
+  }
+
+  if (rollback_images != current_images) {
```

Comment:
This would effectively mean that any older snaps taken before a membership change could not be used for a rollback. Wouldn't that defeat the purpose of having a snap?

Reply:
If the image(s) in question are around, the membership can be trivially restored before issuing the rollback.

Putting the potential data loss aside, conceptually, we want post-rollback group to look exactly the same as it looked when that snapshot was taken.  If the group is going to be missing some images, denying the rollback request seems prudent.